dns_sd_avahi.c

```diff
@@ -58,6 +58,38 @@ void dnssd_free_discovery_data(struct dns_sd_discovery_data *d)
 	free(d);
 }
 
+static void avahi_process_resolved(struct dns_sd_discovery_data *ddata, const AvahiAddress *addr,
+				   const char *host_name, const uint16_t port)
+{
+       /* Avahi is multi-threaded, so lock the list */
+       iio_mutex_lock(ddata->lock);
+       ddata->resolved++;
+
+       /* Find empty data to store things*/
+       while (ddata->next)
+               ddata = ddata->next;
+
+       /* link a new placeholder to the list */
+       avahi_address_snprint(ddata->addr_str,
+                       sizeof(ddata->addr_str), addr);
+       memcpy(ddata->address, addr, sizeof(*ddata->address)); /* Flawfinder: ignore */
+       ddata->port = port;
+       ddata->hostname = strdup(host_name);
```

Comment:
I believe this should be iio_strdup, but since you didn't modify that code, it can be done in a future commit.